src/glue/jobs/s3_to_json_s3.py

```diff
@@ -397,8 +400,11 @@ def validate_against_schema(data, schema):
     # This is a workaround for this bug
     # https://github.com/python-jsonschema/jsonschema/issues/1012
     if "$id" in schema and schema["$id"].startswith("schemas/v0/"):
-        schema["$id"] = ""
-    validator = validator_cls(schema=schema)
+        substitute_schema = copy.deepcopy(schema)
+        substitute_schema["$id"] = ""
+        validator = validator_cls(schema=substitute_schema)
+    else:
+        validator = validator_cls(schema=schema)
```

Comment:
This is the bug fix. This function referenced a pointer to the schema in `json_schemas`, so when I overwrote `$id` I was overwriting the actual schema in `json_schemas`, and not just a local copy. So when it came time to look up the dataset identifier, `$id` was blank. I explicitly make a copy here so that we don't have this problem.

Reply:
I might have asked before, but why are you writing the schema as a blank string?